cli/npm/managed/mod.rs

```diff
@@ -548,35 +539,6 @@ impl CliNpmResolver for ManagedCliNpmResolver {
     self
   }
 
-  fn clone_snapshotted(&self) -> Arc<dyn CliNpmResolver> {
-    // create a new snapshotted npm resolution and resolver
-    let npm_resolution = Arc::new(NpmResolution::new(
-      self.api.clone(),
-      self.resolution.snapshot(),
-      self.maybe_lockfile.clone(),
-    ));
-
-    Arc::new(ManagedCliNpmResolver::new(
-      self.api.clone(),
-      self.fs.clone(),
-      npm_resolution.clone(),
-      create_npm_fs_resolver(
-        self.fs.clone(),
-        self.global_npm_cache.clone(),
-        &self.progress_bar,
-        self.api.base_url().clone(),
-        npm_resolution,
-        self.root_node_modules_path().map(ToOwned::to_owned),
-        self.npm_system_info.clone(),
-      ),
-      self.global_npm_cache.clone(),
-      self.maybe_lockfile.clone(),
-      self.package_json_deps_installer.clone(),
-      self.progress_bar.clone(),
-      self.npm_system_info.clone(),
-    ))
-  }
```

Comment:
The npm cache may be modified when a request is in flight. We should probably still snapshot it? I'm not sure.

Reply:
I think in practice, the isolate's resolver/loader caches are fully populated while processing a completion request or something in the language server lock. Diagnostic requests can happen in parallel with an update but those are queued up after a 200ms delay.

Reply:
Restored this and added `LspResolver::snapshot()`.